### PR TITLE
Separate panel scene

### PR DESCRIPTION
**Pull request by CurtisGreen**
1) Moved code related to the in-game UI (skills, sound, etc.) into a new file called dashboard.js. Creates a separate parallel scene so it can manage loading its own assets
2) Moved calcLevel under constants.js
3) Fixed issues with shop loading

### CODE REVIEW

File: src/resource.js [hunk 1]
```diff
@@ -49,13 +49,13 @@ export class Resource {
...
-        let curLv = this.scene.calcLevel(curXp);
+        let curLv = calcLevel(curXp);
 
         // Increase xp
-        this.scene.updateSkillsText();
+        this.scene.dashboard.updateSkillsText();
 
         // Increase progress and check completion
-        let completed = this.progressBar.updateProgress(curLv);
+        let completed = this.progressBar.updateProgress(curLv);q
```
Comment (CurtisGreen) [inline src/resource.js]: woops gimme a sec

Comment (CurtisGreen) [reply src/resource.js]: Got it

File: src/constants/constants.js [hunk 2]
```diff
@@ -71,3 +72,13 @@ export const MAP = {
         backgroundColor: "grey"
     }
 };
+
+export const calcLevel = function(xp, lv = 1) {
```
Comment (EthanVieira) [inline src/constants/constants.js:81]: Not sure if this is the appropriate place for this. Should we make a utils file for misc methods? Thoughts?

File: src/scenes/enemy-level.js [hunk 3]
```diff
@@ -172,14 +177,28 @@ export class EnemyLevelScene extends LevelScene {
...
                 break;
             case CONSTANTS.CLASS.WARRIOR:
-                return this.calcLevel(this.characterData.skills.attack);
+                return calcLevel(this.characterData.skills.attack);
                 break;
         }
     }
+
+    // Show/hide text
+    showQuestText(show) {
```
Comment (EthanVieira) [inline src/scenes/enemy-level.js]: NIT: can we change the parameter name in both of these methods to isShown or something more indicative of a boolean?

File: src/ui/dashboard.js [hunk 4]
```diff
@@ -0,0 +1,372 @@
...
+                audioButtonRow.push(audioButton);
+            }
+            // Save 2d array of buttons (3 x 5)
+            this.audio.audioButtons.push(audioButtonRow);
+        }
+        // Hide audio page on startup
+        this.showAudioSettings(false);
+    }
+
+    showSkills(show) {
```
Comment (EthanVieira) [inline src/ui/dashboard.js]: isShown for these as well

**Review by EthanVieira — CHANGES_REQUESTED** [review]

File: src/constants/constants.js (hunk 2 above)
Comment (CurtisGreen) [reply src/constants/constants.js:81]: Yeah that sounds good but I'm not sure what else would go there as of now

Comment (CurtisGreen) [reply src/constants/constants.js:81]: Gonna wait til we have more related utilities then I'll separate it out.

File: src/scenes/enemy-level.js (hunk 3 above)
Comment (CurtisGreen) [reply src/scenes/enemy-level.js]: Got it

File: src/ui/dashboard.js (hunk 4 above)
Comment (CurtisGreen) [reply src/ui/dashboard.js]: Done

**Review by EthanVieira — APPROVED** [review]

**Pull request merged**